src/NuGet.Core/NuGet.ProjectModel/LockFile/AssetsLogMessage.cs

```diff
@@ -54,6 +70,11 @@ public AssetsLogMessage(LogLevel logLevel, NuGetLogCode errorCode,
                     targetGraph
                 };
             }
+
+            if (logLevel == LogLevel.Warning)
+            {
+                WarningLevel = WarningLevel.Severe; // setting default to Severe as 0 implies show no warnings
```

Comment:
this `logLevel == LogLevel.Warnings` condition in order to set `WarningLevel`'s default looks like a behaviour change from the old code, and it's not intuitive to me why the condition is needed.